### PR TITLE
coder: add additional build flags

### PR DESCRIPTION
**Pull request by aslilac**
- mark the build as AGPL licensed
- mark the build as a "slim" build (meaning, it does not include the web frontend or embedded binaries)

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/c/coder.rb [hunk 1]
```diff
@@ -21,13 +21,17 @@ def install
     ldflags = %W[
       -s -w
       -X github.com/coder/coder/v2/buildinfo.tag=#{version}
+      -X github.com/coder/coder/v2/buildinfo.agpl=true
```
Comment (SMillerDev) [inline Formula/c/coder.rb:24]: What does this actually do?

Comment (aslilac) [reply Formula/c/coder.rb:24]: `coder version` outputs some licensing information, and it uses this variable to determine which license is in use.

Comment (aslilac) [reply Formula/c/coder.rb:24]: https://github.com/coder/coder/blob/main/scripts/build_go.sh#L110

**Review by chenrui333 — APPROVED** [review]

**Review by p-linnane — APPROVED** [review]

**Review by SMillerDev — APPROVED** [review]

**Review by p-linnane — APPROVED** [review]

**Comment by p-linnane** [comment]
@aslilac Please do not push to PR's once the bottle commit has been added. This breaks our merge process.

**Comment by p-linnane** [comment]
I've rebased to fix the conflict and will let new bottles build and publish.

**Comment by p-linnane** [comment]
> sorry, I was just hoping fixing the conflict would make it rebuild the bottles

No problem. Thanks for contributing to Homebrew!

**Pull request merged**